controllers/mdutil/util.go

```diff
@@ -521,7 +536,7 @@ func NewMSNewReplicas(deployment *clusterv1.MachineDeployment, allMSs []*cluster
 			return 0, err
 		}
 		// Find the total number of machines
-		currentMachineCount := GetReplicaCountForMachineSets(allMSs)
+		currentMachineCount := GetTotalReplicaCountForMachineSets(allMSs)
 		maxTotalMachines := *(deployment.Spec.Replicas) + int32(maxSurge)
 		if currentMachineCount >= maxTotalMachines {
 			// Cannot scale up.
```

Comment:
As an aside, it might be good to see if we can find a way to avoid the duplicated logic for the `default` case below, I'm not sure if it's still the case but I suspect it was there to handle the case where a user didn't define the strategy and we expected it to default to the same logic as RollingUpdateMachineDeploymentStrategyType

Reply:
looks like we already have webhooks to set the defaults if no strategy provided. maybe we can safely skip the defaults here?  https://github.com/kubernetes-sigs/cluster-api/blob/master/api/v1alpha3/machinedeployment_webhook.go#L135

Reply:
+1, thanks for verifying.